src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -74,56 +75,65 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
+                    var type = typeIterator;
 
-                    if (type == null)
+                    // If we have a Nullable<T>, unwrap it.
+                    if (type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
                     {
-                        return;
+                        type = type.GetTypeArguments().FirstOrDefault();
+
+                        if (type == null)
+                        {
+                            continue;
+                        }
                     }
-                }
 
-                if (!type.IsEditorBrowsable(options.GetOption(CompletionOptions.HideAdvancedMembers, semanticModel.Language), semanticModel.Compilation))
-                {
-                    return;
-                }
+                    if (type.TypeKind != TypeKind.Enum)
+                    {
+                        type = TryGetEnumTypeInEnumInitializer(semanticModel, token, type, cancellationToken) ??
+                               TryGetCompletionListType(type, semanticModel.GetEnclosingNamedType(position, cancellationToken), semanticModel.Compilation);
 
-                // Does type have any aliases?
-                var alias = await type.FindApplicableAlias(position, semanticModel, cancellationToken).ConfigureAwait(false);
+                        if (type == null)
+                        {
+                            continue;
+                        }
+                    }
 
-                var displayService = document.GetLanguageService<ISymbolDisplayService>();
-                var displayText = alias != null
-                    ? alias.Name
-                    : displayService.ToMinimalDisplayString(semanticModel, position, type);
+                    if (!type.IsEditorBrowsable(options.GetOption(CompletionOptions.HideAdvancedMembers, semanticModel.Language), semanticModel.Compilation))
+                    {
+                        continue;
+                    }
 
-                var workspace = document.Project.Solution.Workspace;
-                var text = await semanticModel.SyntaxTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                    // Does type have any aliases?
+                    var alias = await type.FindApplicableAlias(position, semanticModel, cancellationToken).ConfigureAwait(false);
 
-                var item = SymbolCompletionItem.CreateWithSymbolId(
-                    displayText: displayText,
-                    displayTextSuffix: "",
-                    symbols: ImmutableArray.Create(alias ?? type),
-                    rules: s_rules.WithMatchPriority(MatchPriority.Preselect),
-                    contextPosition: position);
+                    var displayService = document.GetLanguageService<ISymbolDisplayService>();
+                    var displayText = alias != null
```

Comment:
nit: `alias?.Name ?? displayService.ToMinimalDisplayString(semanticModel, position, type)`

Reply:
Agree. Let us make this change the next time.

---
In reply to: [292627140](https://github.com/dotnet/roslyn/pull/36293#discussion_r292627140) [](ancestors = 292627140)